examples/parquet_write.rs

```diff
@@ -0,0 +1,53 @@
+use std::fs::File;
+use std::iter::once;
+
+use arrow2::{
+    array::{Array, Int32Array},
+    datatypes::Field,
+    error::Result,
+    io::parquet::write::{
+        array_to_page, to_parquet_type, write_file, CompressionCodec, SchemaDescriptor,
+    },
+};
+
+fn write_single_array(path: &str, array: &dyn Array, field: &Field) -> Result<()> {
+    // Create a new empty file
+    let mut file = File::create(path)?;
+
+    // convert arrows' logical type into a parquet type.
+    let parquet_type = to_parquet_type(&field)?;
+
+    let compression = CompressionCodec::Uncompressed;
+
+    // Declare the row groups iterator. This must be an iterator of iterators of iterators:
+    // * first iterator of row groups
+    // * second iterator of column chunks
+    // * third iterator of pages
+    // an array can always be divided in multiple pages via `.slice(offset, length)` (`O(1)`).
+    // All column chunks within a row group MUST have the same length.
+    let row_groups = once(Result::Ok(once(Result::Ok(once(array_to_page(
```

Comment:
The difference is just about the lifetimes. With a vector of multiple pages / groups / etc, all must be pre-allocated and only at the very end they are deallocated. With an iterator, items are deallocated immediately after being consumed, which reduces the memory footprint.

Reply:
I see. That is a very efficient way of using memory. One more reason why I like iterators a lot 😄 

I was asking about the vector replacement because I have found in a lot of code base that not everybody uses iterators and favours the collected form of the iterator. It would make sense to show how to create the iterator the function needs using vectors as well.

Reply:
:) I see it mostly about tradeoffs. E.g. an iterator 

* does not support random access; it must be iterated item by item
* is has an unknown size

So, it is not always the best construct. I decided to use them because:
* we do not need random access
* people may not want to store all pages, chunks etc, as they can be large in size

It is still possible: just allocate the vector and then convert them to an iterator or something.